examples/onnxruntime/quantization/image-classification/run_image_classification.py

```diff
@@ -0,0 +1,434 @@
+    else:
+        data_files = {}
+        if data_args.train_dir is not None:
+            data_files["train"] = os.path.join(data_args.train_dir, "**")
+        if data_args.validation_dir is not None:
+            data_files["validation"] = os.path.join(data_args.validation_dir, "**")
+        dataset = load_dataset(
+            "imagefolder",
+            data_files=data_files,
+            cache_dir=model_args.cache_dir,
+            task="image-classification",
+        )
+        # See more about loading custom images at
+        # https://huggingface.co/docs/datasets/v2.0.0/en/image_process#imagefolder.
+
+    # Prepare label mappings.
+    # We'll include these in the model's config to get human readable labels in the Inference API.
+    labels = dataset["train"].features["labels"].names
+    label2id = {label: str(i) for i, label in enumerate(labels)}
+    id2label = {str(i): label for i, label in enumerate(labels)}
+
+    # Load pretrained model and feature extractor
+    config = AutoConfig.from_pretrained(
+        model_args.model_name_or_path,
+        num_labels=len(labels),
+        i2label=id2label,
+        label2id=label2id,
+        finetuning_task="image-classification",
+    )
+    feature_extractor = AutoFeatureExtractor.from_pretrained(model_args.model_name_or_path)
+    model = AutoModelForImageClassification.from_pretrained(
+        model_args.model_name_or_path,
+        from_tf=bool(".ckpt" in model_args.model_name_or_path),
+        config=config,
+        ignore_mismatched_sizes=model_args.ignore_mismatched_sizes,
+    )
+
+    # Define torchvision transforms to be applied to each image.
+    normalize = Normalize(mean=feature_extractor.image_mean, std=feature_extractor.image_std)
+    transforms = Compose(
+        [
+            Resize(feature_extractor.size),
+            CenterCrop(feature_extractor.size),
+            ToTensor(),
+            normalize,
+        ]
+    )
+
+    def preprocess_function(example_batch):
+        """Apply transforms across a batch."""
+        example_batch["pixel_values"] = [
+            transforms(image.convert("RGB")).to(torch.float32).numpy() for image in example_batch["image"]
+        ]
+        return example_batch
+
+    metric = load_metric("accuracy")
+
+    # You can define your custom compute_metrics function. It takes an `EvalPrediction` object (a namedtuple with a
+    # predictions and label_ids field) and has to return a dictionary string to float.
+    def compute_metrics(p: EvalPrediction):
+        preds = p.predictions[0] if isinstance(p.predictions, tuple) else p.predictions
+        preds = np.argmax(preds, axis=1)
+
+        result = metric.compute(predictions=preds, references=p.label_ids)
+        return result
+
+    # Create the quantizer
+    quantizer = ORTQuantizer(
+        preprocessor=feature_extractor, model=model, feature="image-classification", opset=optim_args.opset
+    )
```

Comment:
I think we should rather push the usage of `ORTQuantizer.from_pretrained` no?

Reply:
There was the argument `ignore_mismatched_sizes` when loading the model in the transformers example, so that's why I loaded the model and quantizer manually. Following text-classification (where there is the same arg in transformers examples but not in Optimum), I removed it and used `ORTQuantizer.from_pretrained`.